rwatch/ui/action_menu.c

```diff
@@ -51,7 +51,7 @@ void * action_menu_item_get_action_data(const ActionMenuItem * item)
 ActionMenuLevel * action_menu_level_create(uint16_t max_items)
 {
     ActionMenuLevel *level = (ActionMenuLevel *)app_calloc(1, sizeof(ActionMenuLevel) + (sizeof(ActionMenuItem) * max_items));
-    level->capacity = max_items;
+    level->capacity = &max_items;
```

Comment:
Now this is bound to break as `max_items` is allocated on the stack and therefore its address will be invalid as soon as one leave the function. This isn't a problem by your cleanup, but a already existing problem (it seems in the whole file)
This should be addressed (very soon!) in an own change.

Reply:
Yes, This doesn't feel right at all. capacity should be a uint16_t and not require a pointer ref. It should be passed by value.